docs/using-the-sdk/cosmovisor.md

```diff
@@ -0,0 +1,174 @@
+directory layout:
+
+```
+.gaiad
+├── config
+├── data
+└── cosmovisor
+```
+
+## Usage
+
+The system administrator admin is responsible for:
+* installing the `cosmovisor` binary and configure the host's init system (e.g. `systemd`, `launchd`, etc) along with the environmental variables appropriately;
+* installing the `genesis` folder manually;
+* installing the `upgrades/<name>` folders manually.
+
+`cosmovisor` will set the `current` link to point to `genesis` at first start (when no `current` link exists) and handles
+binaries switch overs at the correct points in time, so that the system administrator can prepare days in advance and relax at upgrade time.
+
+Note that blockchain applications that wish to support upgrades may package up a genesis `cosmovisor` tarball with this information,
+just as they prepare the genesis binary tarball. In fact, they may offer a tarball will all upgrades up to current point for easy download
+for those who wish to sync a fullnode from start.
+
+The `DAEMON` specific code and operations (e.g. tendermint config, the application db, syncing blocks, etc) are performed as normal.
+Application binaries' directives such as command-line flags and environment variables work normally.
+
+## Example: simd
+
+The following instructions provide a demonstration of `cosmovisor`'s integration with the `simd` application
+shipped along the Cosmos SDK's source code.
+
+First compile `simd`:
+
+```
+cd cosmos-sdk/
+make build
+```
+
+Create a new key and setup the `simd` node:
+
+```
+rm -rf $HOME/.simapp
+./build/simd keys --keyring-backend=test add validator
+./build/simd init testing --chain-id test
+./build/simd add-genesis-account --keyring-backend=test $(./build/simd keys --keyring-backend=test show validator -a) 1000000000stake,1000000000validatortoken
+./build/simd gentx validator --chain-id test
+./build/simd collect-gentxs
+```
+
+Set the required environment variables:
+
+```
+export DAEMON_NAME=simd         # binary name
+export DAEMON_HOME=$HOME/.simapp  # daemon's home directory
+```
+
+Create the `cosmovisor`’s genesis folders and deploy the binary:
+
+```
+mkdir -p $DAEMON_HOME/cosmovisor/genesis/bin
+cp ./build/simd $DAEMON_HOME/cosmovisor/genesis/bin
+```
+
+For the sake of this demonstration, we would amend `voting_params.voting_period` in `.simapp/config/genesis.json` to a reduced time ~5 minutes (300s) and eventually launch `cosmosvisor`:
+
+```
+cosmovisor start
+```
+
+Submit a software upgrade proposal:
+
+```
+./build/simd tx gov submit-proposal software-upgrade test1 --title "upgrade-demo" --description "upgrade"  --from validator --upgrade-height 100 --deposit 10000000stake -y --chain-id test
+```
+ 
+Query the proposal to ensure it was correctly broadcast and added to a block:
+
+```
+./build/simd query gov proposal 1
+```
+ 
+Submit a `Yes` vote for the upgrade proposal:
+
+```
+./build/simd tx gov vote 1 yes --from validator -y --chain-id test
```

Comment:
nit: This needs the `--keyring-backend test` flag to be consistent with the `add-genesis-account` command

Reply:
fixed, please check again

Reply:
ace :bow: